### PR TITLE
Feature test class

### PR DESCRIPTION
**Pull request by veselina-velkova**

### CODE REVIEW

File: my_class.h [hunk 1]
```diff
@@ -12,30 +6,33 @@
...
+   float sum{};
+   std::vector<float> grades{};
+   bool isValidName(const std::string& new_name);
+   bool isValidGrade(const float grade);
 
-    public:
-    student();
-    student(int num, const std::string& full_name);
+   public:
+   //student();
```
Comment (ypetkova93) [inline my_class.h]: remove if not needed?

File: my_class.cpp [hunk 2]
```diff
@@ -1,151 +1,175 @@
...
 
-student::student(int num, const std::string& full_name)
+bool student::isValidName(const std::string& new_name)
 {
-    number = num;
-    name = full_name;
-    grades.resize(MIN_GRADES);
+   for(char ch : new_name)
+   {
+      if(ch != ' ' && ch != '-' && !isalpha(ch))
```
Comment (ypetkova93) [inline my_class.cpp]: I might be paranoid, but I would put explicit brackets around each of these, like this:
if((ch != ' ') && (ch != '-') && !isalpha(ch))

File: test_my_class.cpp [hunk 3]
```diff
@@ -1,47 +1,880 @@
...
+
+#define TEST_CLASS_NOT_THAT_LARGE_POS_INT (31650)
+#define TEST_CLASS_NOT_THAT_LARGE_NEGATIVE_INT (-31650)
+
+#define TEST_CLASS_LARGE_POSITIVE_INT (std::numeric_limits<int>::max())
+#define TEST_CLASS_LARGE_NEGATIVE_INT (std::numeric_limits<int>::min())
 
-#define TEST_CLASS_POSITIVE_FLOAT(5.5)
-#define TEST_CLASS_NEGATIVE_FLOAT(-3.5)
+#define TEST_CLASS_UNSIGNED_LONG_LONG_INT (std::numeric_limits<unsigned long long int>::max())
```
Comment (ypetkova93) [inline test_my_class.cpp:29]: stupid qn: what's wrong with INT_MAX?
this is a LONG LONG INT but let's say on line 26, why not use INT_MAX?

File: test_my_class.cpp [hunk 4]
```diff
@@ -1,47 +1,880 @@
...
+   //UT: Testing if getMaxGrade works for student who has at least 1 grade.
+   //We will get random number of grades and random grades in the wanted scope.
+   grades.clear();
+   int count = 1 + (rand() / (RAND_MAX / (MAX_GRADES - 1)));
+   for(int i = test_student.getNumberGrades(); i < count; ++i)
+   {
+      float new_grade = LOWEST_GRADE + (rand()/(RAND_MAX / (HIGHEST_GRADE - LOWEST_GRADE)));
+      check_addNewGrade(test_student, new_grade);
+      grades.push_back(new_grade);
+   }
```
Comment (ypetkova93) [inline test_my_class.cpp:714]: lines 713 - 720 repeat in different functions. Suggestion: create a new function that takes in "student" type, randomly creates the grades as you're doing, and returns a ptr to the object (or you can pass the students by reference, as we learned in Zhivko's course).

that way you can replace the code with a one liner call to a function on lines 773-779, 833-838, and anywhere else if I missed it..

File: test_my_class.cpp [hunk 5]
```diff
@@ -1,47 +1,880 @@
...
+   name = TEST_CLASS_VALID_STRING;
+   number =TEST_CLASS_POSITIVE_INT;
+   student test_student(number, name);
+
+   //UT: Check when no grades are received.
+   grades.clear();
+   check_are_recieved_grades_the_same(test_student, grades);
 }
 
+void TestClass :: check_are_recieved_grades_the_same(student& test_student, std::vector<float>& temp_grades)
```
Comment (ypetkova93) [inline test_my_class.cpp]: received* typo, not a big deal though

File: test_my_class.cpp (hunk 3 above)
Comment (veselina-velkova) [reply test_my_class.cpp:29]: Oh I think it should work as well, but unsigned long long int has max value 18, 446, 744, 073, 709, 551, 615 and INT_MAX (64bit integer) is 9,223,372,036,854,775,807.

File: my_class.h (hunk 1 above)
Comment (veselina-velkova) [reply my_class.h]: Here you mean the comment?

Comment (ypetkova93) [reply my_class.h]: yes, the comment

File: test_my_class.cpp (hunk 3 above)
Comment (ypetkova93) [reply test_my_class.cpp:29]: ok, fair enough. I have been using INT_MAX for int types, but the numerical limits also work.

**Review by ypetkova93 — COMMENTED** [review]
Good job!

**Comment by veselina-velkova** [comment]
> Good job!

Thanks, Yoli! :)

**Pull request merged**